### PR TITLE
Pre-release enhancements

### PR DESCRIPTION
**Pull request by sarah-tuva**
## Describe your changes
Several enhancements were made to improve the organization and functionality of data profiling. Major changes called out:

* Models:
   - Base models added for claim sources with logic to use an empty table when the source is missing
   - Pharmacy claims added to data profiling checks
   - Renamed and reordered tests in the detail and summary models
 * Seeds:
   - Added test_catalog with metadata for every data profiling check
 * Packages:
   - Using surrogate key macro from dbt_utils
 * Macros:
   - Improved date check macros to use try cast and safe cast methods
   - Added a new macro to cast string or varchar depending on adapter type

## Validation & Testing
* Tested on three different sources in Snowflake and BigQuery
* Tested code snippets in Redshift

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] My code follows style guidelines
- [x] I have commented my code as necessary
- [x] I have made corresponding changes to documentation
- [x] My code runs without errors
- [x] I have implemented primary key tests
- [x] Every model has a corresponding yml file
- [x] I have generated the DAG locally with dbt and updated the files in the `docs/` path

### CODE REVIEW

File: models/claims/claim_summary.sql [hunk 1]
```diff
@@ -1,202 +0,0 @@
-{#-
```
Comment (sarah-tuva) [inline models/claims/claim_summary.sql:1]: The "claims" folder was renamed to "claims_input_layer" for clarity.

File: models/claims_input_layer/base/base_eligibility.sql [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                , 'first_name'
+                , 'last_name'
+                , 'address'
+                , 'city'
+                , 'state'
+                , 'zip_code'
+                , 'phone'
+                , 'data_source'
+               ]) }}
+           as row_hash
```
Comment (sarah-tuva) [inline models/claims_input_layer/base/base_eligibility.sql:76]: This is used for joining in the detail models to ensure we join on exact records when there are duplicates.

File: models/claims_input_layer/claim_summary.sql [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+  )
+}}
+{%- elif target.type in ('bigquery', 'snowflake') -%}
+{{
+  config(
+    cluster_by = ['test_id']
+  )
+}}
+{%- else -%}
+{%- endif -%}
```
Comment (sarah-tuva) [inline models/claims_input_layer/claim_summary.sql:14]: Logic added to order results in final summary table.

File: snapshots/snapshot_claim_summary.sql [hunk 4]
```diff
@@ -6,7 +6,7 @@
       , target_schema = var('output_schema')
       , strategy = 'timestamp'
       , updated_at = 'run_date'
-      , unique_key = "table_name||'-'||test_name"
+      , unique_key = 'test_name||run_date'
```
Comment (sarah-tuva) [inline snapshots/snapshot_claim_summary.sql:9]: Removed "-" as it was causing data type issues and added run_date to ensure uniqueness due to an inconsistent bug with dbt.

**Review by aneiderhiser — APPROVED** [review]

**Pull request merged**